src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -473,6 +475,26 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 			}
 			return findWidget;
 		});
+		const viewKey = new Set<string>();
+		viewKey.add('focusedView');
+		TabFocus.onDidChangeTabFocus(tabFocus => {
+			if (contextKeyService.getContextKeyValue('focusedView') === 'terminal') {
+				this._terminalTabFocus = tabFocus;
+			} else {
+				this._editorTabFocus = tabFocus;
+			}
+		});
+		this._register(contextKeyService.onDidChangeContext((c) => {
+			if (c.affectsSome(viewKey)) {
+				if (contextKeyService.getContextKeyValue('focusedView') === 'terminal') {
+					this._editorTabFocus = TabFocus.getTabFocusMode();
+					TabFocus.setTabFocusMode(this._terminalTabFocus ?? this._configurationService.getValue(TerminalSettingId.TabFocusMode));
+				} else {
+					this._terminalTabFocus = TabFocus.getTabFocusMode();
+					TabFocus.setTabFocusMode(this._editorTabFocus ?? this._configurationService.getValue('editor.tabFocusMode'));
+				}
+			}
+		}));
```

Comment:
nit: This is something we'll eventually want to move out of TerminalInstance, I'm not sure the best place for it but I think it's all standalone so it could live in `TerminalMainContribution` for example.

Reply:
yeah, wasn't sure where to put it. that works